### ISSUE TITLE
RPCBind Package

### ISSUE THREAD

**Issue by CamoMacdonald** [vmware/photon#736]
When trying to install the RPCBind package - this seems to randomly fail. We have 2 identical VMs one installed correctly the other didn't.

```
root [ / ]# tdnf install rpcbind

Installing:
rpcbind     x86_64   0.2.4-4.ph2    photon    83.88k 85893

Total installed size:  83.88k 85893
Is this ok [y/N]:y

Downloading:
rpcbind                                  54151    100%
Testing transaction
Running transaction
Error(-1) : Unknown error -1601
```

I can see this was a previous issue with the docker setup however seemed to have been resolved since then.

**Comment by ppadmavilasom** [comment]
what version of photon os (/etc/os-release)
is this inside a docker container?
can you please use "tdnf install rpcbind --rpmverbosity debug" and post the relevant results?

**Comment by CamoMacdonald** [comment]
I tested this on both photon 1.0 and 2.0. This was on the host its self (VM) not within a docker container
See below for debug:

```
Downloading:
rpcbind                                  54151    100%
D: Expected size:        54151 = lead(96)+sigs(4660)+pad(4)+data(49391)
D:   Actual size:        54151
D: /var/cache/tdnf/photon/rpms/x86_64/rpcbind-0.2.4-4.ph2.x86_64.rpm: Header SHA1 digest: OK (693fb0c6edeba80db99f9f81b9276d3f84d61e7d)
fdio:       6 reads,    11318 total bytes in 0.000009 secs
D: loading keyring from pubkeys in /var/lib/rpm/pubkeys/*.key
D: couldn't find any keys in /var/lib/rpm/pubkeys/*.key
D: loading keyring from rpmdb
D: opening  db environment /var/lib/rpm cdb:0x401
D: opening  db index       /var/lib/rpm/Packages 0x400 mode=0x0
D: opening  db index       /var/lib/rpm/Name 0x400 mode=0x0
D:  read h#       6 Header SHA1 digest: OK (8cee1a0443b8f1fa131539392590307ed758767e)
D: added key gpg-pubkey-66fd4949-4803fe57 to keyring
D: Using legacy gpg-pubkey(s) from rpmdb
D: Expected size:        54151 = lead(96)+sigs(4660)+pad(4)+data(49391)
D:   Actual size:        54151
D: /var/cache/tdnf/photon/rpms/x86_64/rpcbind-0.2.4-4.ph2.x86_64.rpm: Header V3 RSA/SHA1 Signature, key ID 66fd4949: OK
D: Plugin: calling hook init in syslog plugin
ufdio:       6 reads,    11318 total bytes in 0.000006 secs
D: ========== recording tsort relations
D: ========== tsorting packages (order, #predecessors, #succesors, depth)
D:     0    0    0    1   +rpcbind-0.2.4-4.ph2.x86_64
D: opening  db index       /var/lib/rpm/Conflictname 0x400 mode=0x0
D: opening  db index       /var/lib/rpm/Requirename 0x400 mode=0x0
D: ========== +++ rpcbind-0.2.4-4.ph2 x86_64/linux 0x2
D: opening  db index       /var/lib/rpm/Basenames 0x400 mode=0x0
D:  read h#      16 Header V3 RSA/SHA1 Signature, key ID 66fd4949: OK
D:  Requires: /bin/chown                                    YES (db files)
D:  Requires: /bin/false                                    YES (db files)
D:  read h#      91 Header V3 RSA/SHA1 Signature, key ID 66fd4949: OK
D:  Requires: /bin/sh                                       YES (db files)
D:  Requires: /bin/sh                                       YES (cached)
D:  Requires: /bin/sh                                       YES (cached)
D:  Requires: /bin/sh                                       YES (cached)
D:  read h#      40 Header V3 RSA/SHA1 Signature, key ID 66fd4949: OK
D:  Requires: /usr/sbin/groupadd                            YES (db files)
D:  Requires: /usr/sbin/groupdel                            YES (db files)
D:  Requires: /usr/sbin/groupdel                            YES (cached)
D:  Requires: /usr/sbin/useradd                             YES (db files)
D:  Requires: /usr/sbin/userdel                             YES (db files)
D:  Requires: /usr/sbin/userdel                             YES (cached)
D:  Requires: config(rpcbind) = 0.2.4-4.ph2                 YES (added provide)
D: opening  db index       /var/lib/rpm/Providename 0x400 mode=0x0
D:  read h#      14 Header V3 RSA/SHA1 Signature, key ID 66fd4949: OK
D:  Requires: libc.so.6()(64bit)                            YES (db provides)
D:  Requires: libc.so.6(GLIBC_2.14)(64bit)                  YES (db provides)
D:  Requires: libc.so.6(GLIBC_2.2.5)(64bit)                 YES (db provides)
D:  Requires: libc.so.6(GLIBC_2.3)(64bit)                   YES (db provides)
D:  Requires: libc.so.6(GLIBC_2.3.4)(64bit)                 YES (db provides)
D:  Requires: libc.so.6(GLIBC_2.4)(64bit)                   YES (db provides)
D:  Requires: libpthread.so.0()(64bit)                      YES (db provides)
D:  Requires: libpthread.so.0(GLIBC_2.2.5)(64bit)           YES (db provides)
D:  read h#      30 Header V3 RSA/SHA1 Signature, key ID 66fd4949: OK
D:  Requires: libsystemd.so.0()(64bit)                      YES (db provides)
D:  Requires: libsystemd.so.0(LIBSYSTEMD_209)(64bit)        YES (db provides)
D:  read h#     106 Header V3 RSA/SHA1 Signature, key ID 66fd4949: OK
D:  Requires: libtirpc                                      YES (db provides)
D:  Requires: libtirpc.so.3()(64bit)                        YES (db provides)
D:  Requires: libtirpc.so.3(TIRPC_0.3.0)(64bit)             YES (db provides)
D:  Requires: libtirpc.so.3(TIRPC_0.3.3)(64bit)             YES (db provides)
D:  Requires: libtirpc.so.3(TIRPC_PRIVATE)(64bit)           YES (db provides)
D:  Requires: rpmlib(CompressedFileNames) <= 3.0.4-1        YES (rpmlib provides)
D:  Requires: rpmlib(PayloadFilesHavePrefix) <= 4.0-1       YES (rpmlib provides)
D:  Requires: systemd                                       YES (db provides)
D: opening  db index       /var/lib/rpm/Obsoletename 0x400 mode=0x0
Testing transaction
D: sanity checking 1 elements
D: Plugin: calling hook tsm_pre in syslog plugin
D: computing 9 file fingerprints
D: opening  db index       /var/lib/rpm/Group 0x400 mode=0x0
D: opening  db index       /var/lib/rpm/Triggername 0x400 mode=0x0
D: opening  db index       /var/lib/rpm/Dirnames 0x400 mode=0x0
D: opening  db index       /var/lib/rpm/Installtid 0x400 mode=0x0
D: opening  db index       /var/lib/rpm/Sigmd5 0x400 mode=0x0
D: opening  db index       /var/lib/rpm/Sha1header 0x400 mode=0x0
D: opening  db index       /var/lib/rpm/Filetriggername 0x400 mode=0x0
D: opening  db index       /var/lib/rpm/Transfiletriggername 0x400 mode=0x0
D: opening  db index       /var/lib/rpm/Recommendname 0x400 mode=0x0
D: opening  db index       /var/lib/rpm/Suggestname 0x400 mode=0x0
D: opening  db index       /var/lib/rpm/Supplementname 0x400 mode=0x0
D: opening  db index       /var/lib/rpm/Enhancename 0x400 mode=0x0
D: computing file dispositions
D: 0x00000802     4096     13835978      3712914 /
D: ========== +++ rpcbind-0.2.4-4.ph2 x86_64-linux 0x2
D: Expected size:        54151 = lead(96)+sigs(4660)+pad(4)+data(49391)
D:   Actual size:        54151
D: rpcbind-0.2.4-4.ph2.x86_64: Header V3 RSA/SHA1 Signature, key ID 66fd4949: OK
ufdio:       6 reads,    11318 total bytes in 0.000007 secs
D: Plugin: calling hook tsm_post in syslog plugin
Running transaction
D: closed   db index       /var/lib/rpm/Packages
D: closed   db index       /var/lib/rpm/Enhancename
D: closed   db index       /var/lib/rpm/Supplementname
D: closed   db index       /var/lib/rpm/Suggestname
D: closed   db index       /var/lib/rpm/Recommendname
D: closed   db index       /var/lib/rpm/Transfiletriggername
D: closed   db index       /var/lib/rpm/Filetriggername
D: closed   db index       /var/lib/rpm/Sha1header
D: closed   db index       /var/lib/rpm/Sigmd5
D: closed   db index       /var/lib/rpm/Installtid
D: closed   db index       /var/lib/rpm/Dirnames
D: closed   db index       /var/lib/rpm/Triggername
D: closed   db index       /var/lib/rpm/Obsoletename
D: closed   db index       /var/lib/rpm/Conflictname
D: closed   db index       /var/lib/rpm/Providename
D: closed   db index       /var/lib/rpm/Requirename
D: closed   db index       /var/lib/rpm/Group
D: closed   db index       /var/lib/rpm/Basenames
D: closed   db index       /var/lib/rpm/Name
D: closed   db environment /var/lib/rpm
D: opening  db environment /var/lib/rpm cdb:0x401
D: opening  db index       /var/lib/rpm/Packages (none) mode=0x42
D: sanity checking 1 elements
D: opening  db index       /var/lib/rpm/Name (none) mode=0x42
D: Plugin: calling hook tsm_pre in syslog plugin
D: opening  db index       /var/lib/rpm/Transfiletriggername (none) mode=0x42
D: opening  db index       /var/lib/rpm/Dirnames (none) mode=0x42
D: running pre-transaction scripts
D: computing 9 file fingerprints
D: opening  db index       /var/lib/rpm/Basenames (none) mode=0x42
D: opening  db index       /var/lib/rpm/Group (none) mode=0x42
D: opening  db index       /var/lib/rpm/Requirename (none) mode=0x42
D: opening  db index       /var/lib/rpm/Providename (none) mode=0x42
D: opening  db index       /var/lib/rpm/Conflictname (none) mode=0x42
D: opening  db index       /var/lib/rpm/Obsoletename (none) mode=0x42
D: opening  db index       /var/lib/rpm/Triggername (none) mode=0x42
D: opening  db index       /var/lib/rpm/Installtid (none) mode=0x42
D: opening  db index       /var/lib/rpm/Sigmd5 (none) mode=0x42
D: opening  db index       /var/lib/rpm/Sha1header (none) mode=0x42
D: opening  db index       /var/lib/rpm/Filetriggername (none) mode=0x42
D: opening  db index       /var/lib/rpm/Recommendname (none) mode=0x42
D: opening  db index       /var/lib/rpm/Suggestname (none) mode=0x42
D: opening  db index       /var/lib/rpm/Supplementname (none) mode=0x42
D: opening  db index       /var/lib/rpm/Enhancename (none) mode=0x42
D: computing file dispositions
D: 0x00000802     4096     13835978      3712914 /
D: ========== +++ rpcbind-0.2.4-4.ph2 x86_64-linux 0x2
D: Expected size:        54151 = lead(96)+sigs(4660)+pad(4)+data(49391)
D:   Actual size:        54151
D: rpcbind-0.2.4-4.ph2.x86_64: Header V3 RSA/SHA1 Signature, key ID 66fd4949: OK
D:   install: rpcbind-0.2.4-4.ph2.x86_64 has 9 files
D: %prein(rpcbind-0.2.4-4.ph2.x86_64): scriptlet start
fdio:       2 writes,      333 total bytes in 0.000012 secs
D: %prein(rpcbind-0.2.4-4.ph2.x86_64): execv(/bin/sh) pid 636
++ getent passwd rpc
++ cut -d: -f 3
+ rpcid=
+ '[' -n '' -a '' '!=' 31 ']'
+ '[' -z '' -o '' '!=' 31 ']'
+ groupadd -g 31 rpc
+ useradd -d /var/lib/rpcbind -g rpc -s /bin/false -u 31 rpc
D: %prein(rpcbind-0.2.4-4.ph2.x86_64): waitpid(636) rc 636 status 600
error: %prein(rpcbind-0.2.4-4.ph2.x86_64) scriptlet failed, exit status 6
D: Plugin: calling hook scriptlet_post in syslog plugin
D: Plugin: calling hook psm_post in ima plugin
D: Plugin: calling hook psm_post in syslog plugin
ufdio:       6 reads,    11318 total bytes in 0.000005 secs
error: rpcbind-0.2.4-4.ph2.x86_64: install failed
D: running post-transaction scripts
D: Plugin: calling hook tsm_post in syslog plugin
D: ========== +++ rpcbind-0.2.4-4.ph2 x86_64/linux 0x2
D: closed   db index       /var/lib/rpm/Packages
D: closed   db index       /var/lib/rpm/Enhancename
D: closed   db index       /var/lib/rpm/Supplementname
D: closed   db index       /var/lib/rpm/Suggestname
D: closed   db index       /var/lib/rpm/Recommendname
D: closed   db index       /var/lib/rpm/Transfiletriggername
D: closed   db index       /var/lib/rpm/Filetriggername
D: closed   db index       /var/lib/rpm/Sha1header
D: closed   db index       /var/lib/rpm/Sigmd5
D: closed   db index       /var/lib/rpm/Installtid
D: closed   db index       /var/lib/rpm/Dirnames
D: closed   db index       /var/lib/rpm/Triggername
D: closed   db index       /var/lib/rpm/Obsoletename
D: closed   db index       /var/lib/rpm/Conflictname
D: closed   db index       /var/lib/rpm/Providename
D: closed   db index       /var/lib/rpm/Requirename
D: closed   db index       /var/lib/rpm/Group
D: closed   db index       /var/lib/rpm/Basenames
D: closed   db index       /var/lib/rpm/Name
D: closed   db environment /var/lib/rpm
Error(-1) : Unknown error -1601
```

**Comment by CamoMacdonald** [comment]
Running that script specifically returns the below:
```
 [ ~ ]$ sudo bash -x rpc.sh 
+ cut -d: -f 3
+ rpcid=getent
+ passwd rpc
passwd: user 'rpc' does not exist
+ '[' -n '' -a '' '!=' 31 ']'
+ '[' -z '' -o '' '!=' 31 ']'
+ groupadd -g 31 rpc
+ useradd -d /var/lib/rpcbind -g rpc -s /bin/false -u 31 rpc
```
I cannot get consistent steps to replicate this as we have 2 identical VMs. One that has this issue, the other that doesn't.  They're brand new installs of Photon so shouldn't be anything specific except running makecache and updates then trying to install

**Comment by CamoMacdonald** [comment]
Adding the Group and User manually seems to allow the install to be successful

**Comment by ppadmavilasom** [comment]
>>2 identical VMs 
could you post the install details - iso install? minimal or full?

**Comment by CamoMacdonald** [comment]
To my knowledge this is full install

**Comment by xiaolin-vmware** [comment]
In the pre install scrip, rpcbind package check whether rpc user exits with uid  31. If the user doesn't exists or uid is not 31,  rpcbind try to create rpc group with gid 31, and user with uid 31. If there is other group with gid 31 or other user with uid 31, the pre install script will fail with error message : Error(-1) : Unknown error -1601
I failed to install rpcbind after I created a user aax with uid 31. 
Here is the console output of my machine:
root@photon-aff9c21ce3d4 [ ~ ]# getent passwd 31
aax:x:31:100::/home/aax:/bin/bash
root@photon-aff9c21ce3d4 [ ~ ]# getent group 31
root@photon-aff9c21ce3d4 [ ~ ]# getent group rpc
root@photon-aff9c21ce3d4 [ ~ ]# getent passwd rpc
root@photon-aff9c21ce3d4 [ ~ ]# tdnf install rpcbind

Installing:
rpcbind                  x86_64       0.2.4-4.ph2      photon        83.88k 85893

Total installed size:  83.88k 85893
Is this ok [y/N]:y

Downloading:
rpcbind                                  54151    100%
Testing transaction
Running transaction
Error(-1) : Unknown error -1601
